exchange/exchange_test.go

```diff
@@ -314,51 +314,145 @@ func TestGetBidCacheInfo(t *testing.T) {
+
+	liveAdapters := make([]openrtb_ext.BidderName, 1)
+	liveAdapters[0] = "appnexus"
+
+	bidRequest := &openrtb.BidRequest{
+		ID: "some-request-id",
+		Imp: []openrtb.Imp{{
+			ID:     "some-impression-id",
+			Banner: &openrtb.Banner{Format: []openrtb.Format{{W: 300, H: 250}, {W: 300, H: 600}}},
+			Ext:    json.RawMessage(`{"appnexus": {"placementId": 10433394}}`),
+		}},
+		Site:   &openrtb.Site{Page: "prebid.org", Ext: json.RawMessage(`{"amp":0}`)},
+		Device: &openrtb.Device{UA: "curl/7.54.0", IP: "::1"},
+		AT:     1,
+		TMax:   500,
+		Ext:    json.RawMessage(`{"id": "some-request-id","site": {"page": "prebid.org"},"imp": [{"id": "some-impression-id","banner": {"format": [{"w": 300,"h": 250},{"w": 300,"h": 600}]},"ext": {"appnexus": {"placementId": 10433394}}}],"tmax": 500}`),
 	}
 
-	//resolvedRequest json.RawMessage
 	resolvedRequest := json.RawMessage(`{"id": "some-request-id","site": {"page": "prebid.org"},"imp": [{"id": "some-impression-id","banner": {"format": [{"w": 300,"h": 250},{"w": 300,"h": 600}]},"ext": {"appnexus": {"placementId": 10433394}}}],"tmax": 500}`)
 
-	//adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra,
 	adapterExtra := map[openrtb_ext.BidderName]*seatResponseExtra{
-		bidderName: {
-			ResponseTimeMillis: 5,
-			Errors: []openrtb_ext.ExtBidderError{
+		"appnexus": {ResponseTimeMillis: 5},
+	}
+
+	var errList []error
+
+	sampleBid := &openrtb.Bid{
+		ID:    "some-imp-id",
+		Price: 9.517803,
+		W:     300,
+		H:     250,
+		Ext:   nil,
+	}
+	aPbsOrtbBidArr := []*pbsOrtbBid{{bid: sampleBid, bidType: openrtb_ext.BidTypeBanner}}
+	sampleSeatBid := []openrtb.SeatBid{
+		{
+			Seat: "appnexus",
+			Bid: []openrtb.Bid{
 				{
-					Code:    999,
-					Message: "Post ib.adnxs.com/openrtb2?query1&query2: unsupported protocol scheme \"\"",
+					ID:    "some-imp-id",
+					Price: 9.517803,
+					W:     300,
+					H:     250,
+					Ext:   json.RawMessage(`{"prebid":{"type":"banner"}}`),
 				},
 			},
 		},
 	}
+	emptySeatBid := []openrtb.SeatBid{}
 
-	return liveAdapters, adapterBids, resolvedRequest, adapterExtra
+	// Test cases
+	type aTest struct {
+		description         string
+		adapterBids         map[openrtb_ext.BidderName]*pbsOrtbSeatBid
+		expectedBidResponse *openrtb.BidResponse
+	}
+	testCases := []aTest{
+		{
+			description: "1) Adapter to bids map comes with a non-empty currency field and non-empty bid array",
+			adapterBids: map[openrtb_ext.BidderName]*pbsOrtbSeatBid{
+				openrtb_ext.BidderName("appnexus"): {
+					bids:     aPbsOrtbBidArr,
+					currency: "USD",
+				},
+			},
+			expectedBidResponse: &openrtb.BidResponse{
+				ID:      "some-request-id",
+				SeatBid: sampleSeatBid,
+				Cur:     "USD",
+				Ext: json.RawMessage(`{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}
+`),
```

Comment:
Consider moving these terminal `')' pairs to the end of the line. Might look cleaner.

Reply:
If instead of:
```
    Ext: json.RawMessage(`{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}
`),
```
I put everything in a single line:
```
    Ext:     json.RawMessage(`{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}`),
```
Tht test faisl because we are missing a line break:
```
Messages:       [TEST_FAILED] Objects must be equal for test: 1) Adapter to bids map comes with a non-empty currency field and non-empty bid array
                 Expected: >>{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}<<
                 Actual: >>{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}
                <<
```
I googled this issue and apparently JSON should recognize the common line break escape sequence `\n` and even `\u000A`, but we still fail the test:
```
Messages:       [TEST_FAILED] Objects must be equal for test: 1) Adapter to bids map comes with a non-empty currency field and non-empty bid array
                 Expected: >>{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}\n<<
                 Actual: >>{"responsetimemillis":{"appnexus":5},"tmaxrequest":500}
                <<
```
Have you come accross this before?

Reply:
Ah, ok. The response must include a line break at the end. Got it. That's fine. Please just resolve the merge conflict and we'll be good here.